code/game/objects/items/granters.dm

```diff
@@ -359,6 +359,32 @@
 	user.adjustFireLoss(6)
 	qdel(src)
 
+/obj/item/book/granter/martial/cqm
+	martial = /datum/martial_art/cqm
+	name = "old scroll"
+	martialname = "close quarters mimery"
+	desc = "A scroll that appears blank, but on closer inspection is written in invisible ink."
+	greet = "<span class='sciradio'>You have learned the ancient martial art Close Quarters Mimery! You are now able to use your new abilities in a fight to silence and confuse your foes! \
+	Check the combos you are now able to perform using the Recall Teachings verb in the Close Quarters Mimery tab.</span>"
+	icon = 'icons/obj/wizard.dmi'
+	icon_state = "scroll2"
+	remarks = list("...", "???", "!!!", "?!?", "!?!")
+
+/obj/item/book/granter/martial/cqm/onlearned(mob/living/carbon/user)
+	..()
+	if(!oneuse)
+		return
+	desc = "It's completely blank."
+	name = "empty scroll"
+	icon_state = "blankscroll"
+
+/obj/item/book/granter/martial/cqm/already_known(mob/user)
+	if(user.mind && !user.mind.miming)
+		to_chat(user, "<span class='warning'>You try to read the scroll but can't comprehend any of it.</span>")
+		return TRUE
+	else
+		return FALSE
```

Comment:
This logic is pretty weird. You are returning true if a player does not have mind or is not a mime or else your return false?

Reply:
yeah i just need !user.mind.miming so user.mind is not needed. return true means cannot learn in granters false means can learn i know weird.

Reply:
and yes i tested it... only mimes that have a vow of silence can read it, nobody else